ios/brave-ios/Sources/Brave/Frontend/Browser/New Tab Page/Backgrounds/NewTabPageVideoButtonsView.swift

```diff
@@ -0,0 +1,199 @@
+/// button and handles user tap gestures to play/pause the video.
+class NewTabPageVideoButtonsView: UIView {
+  var tappedBackground: (() -> Bool)?
+  var tappedCancelButton: (() -> Void)?
+
+  private let playButtonImage = BluredImageView(imageName: "ntt_play_button").then {
+    $0.isHidden = true
+    $0.alpha = 0
+  }
+
+  private let pauseButtonImage = BluredImageView(imageName: "ntt_pause_button").then {
+    $0.isHidden = true
+    $0.alpha = 0
+  }
+
+  private let cancelButton = CancelButton().then {
+    $0.isHidden = false
+  }
+
+  init() {
+    super.init(frame: .zero)
+
+    backgroundColor = .clear
+
+    addSubview(cancelButton)
+
+    addSubview(playButtonImage)
+    addSubview(pauseButtonImage)
+
+    let tapGesture: UITapGestureRecognizer = UITapGestureRecognizer(
+      target: self,
+      action: #selector(self.videoTapped(sender:))
+    )
+    tapGesture.numberOfTapsRequired = 1
+    addGestureRecognizer(tapGesture)
+  }
+
+  private func tappedVideoCancelButton() {
+    tappedCancelButton?()
+  }
+
+  @objc private func videoTapped(sender: UITapGestureRecognizer) {
+    let location = sender.location(in: self)
+    if let view = super.hitTest(location, with: nil), view == cancelButton {
+      tappedVideoCancelButton()
+      return
+    }
+
+    guard let playStarted = tappedBackground?() else {
+      return
+    }
+
+    if playStarted {
+      playButtonImage.isHidden = false
+      pauseButtonImage.isHidden = true
+      showAndFadeOutImage(imageView: playButtonImage)
+    } else {
+      pauseButtonImage.isHidden = false
+      playButtonImage.isHidden = true
+      showAndFadeOutImage(imageView: pauseButtonImage)
+    }
+  }
+
+  private func showAndFadeOutImage(imageView: UIView) {
+    imageView.alpha = 0
+    UIView.animate(
+      withDuration: 0.1,
+      animations: {
+        imageView.alpha = 1
+      },
+      completion: { _ in
+        UIView.animate(
+          withDuration: 0.3,
+          delay: 0.5,
+          animations: {
+            imageView.alpha = 0
+          }
+        )
+      }
+    )
+  }
+
+  @available(*, unavailable)
+  required init(coder: NSCoder) {
+    fatalError()
+  }
+
+  override func layoutSubviews() {
+    super.layoutSubviews()
+
+    cancelButton.snp.remakeConstraints {
+      $0.top.equalTo(self.safeAreaLayoutGuide.snp.top).offset(20)
+      $0.right.equalTo(self.safeAreaLayoutGuide.snp.right).offset(-20)
+    }
+
+    playButtonImage.snp.remakeConstraints {
+      $0.centerX.equalToSuperview()
+      $0.centerY.equalToSuperview().offset(20)
+    }
+
+    pauseButtonImage.snp.remakeConstraints {
+      $0.centerX.equalToSuperview()
+      $0.centerY.equalToSuperview().offset(20)
+    }
+  }
+}
+
+extension NewTabPageVideoButtonsView {
+  private class CancelButton: SpringButton {
+    let imageView = UIImageView(
+      image: UIImage(named: "ntt_cancel_button", in: .module, compatibleWith: nil)!
+    )
```

Comment:
Just make this frosted background button a generic button that doesnt have a specific image and use it multiple times setting the image as needed in each spot used